### ISSUE TITLE
API Calls being made when they should not

### ISSUE THREAD

**Issue by myElectrical** [FlutterFlow/flutterflow-issues#2993]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

I have a project with several pages. I have added two API calls, one on one page and the second on another page. I am testing my project using desktop, Chrome, and locally.

First I start/reload the project. If I then navigate between pages without API calls, no API calls are made (as expected).  If I then navigate to a page with an API call, not only is that API call made but the call second call (on another page) is made.  If I then naviage to any page (with or without an API call), both API calls are made.  These calls are made every time I navigate to any page. 

The first API call [Get] is attached to 'Card' widget on that page. The second API call [Put] is attached to an Action connected to a button click. 

As I continue to develop my application, I  will add many API calls, so this issue is very concerning.

### Expected Behavior

I expect the first API call to be invoked only when the Card widget is rendered.  I expect the second API call to be invoked only when the button on the page is clicked. 

### Steps to Reproduce

1. See current behaviour for an explanation of what is happening.
2. I have a ticked-blank project to proceed, but it is a relatively new project I am developing.  A blank project would require login to get jwtToken for the API call. 

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITFTi/KA3It1rs9H+KrAb8dKjicRI0McWOdEl+tuZwgbIoznBqQhPszRM1ZtOce6aVNcemL9gj8C+NbKuOD9GvU4BymBRoRmyLl2cgvke2ilMMmXCs6KZ3ElP59NCm6h5sHYkBR5CMBfW0YW6jmHN96/TgHDFOu/Zwh9f6PHaOI=

### Context

I am migrating my application from Angular.  Having got the boilerplate sorted out, this issue is the only thing which would concern me about progress.  I am worried that with multiple API calls this become app breaking. 

### Visual documentation

![2024-05-27_14-14-15](https://github.com/FlutterFlow/flutterflow-issues/assets/22624191/9cad5635-d82d-475c-8e5a-9d9d37129834)

### Environment

```markdown
- FlutterFlow version: v4.1.53
- Platform: Web
- Browser name and version: Chrome
- Operating system and version affected: Windows 11
```

### General

- [ ] I was collaborating with other users in the same project prior to data loss
- [ ] I navigated between different project views leading up to data loss
- [ ] The UI was running very slowly prior to data loss
- [ ] A crash or other bug occurred prior to data loss
- [ ] Data was not saved in a snapshot
- [ ] I was using flutterflow from multiple devices
- [ ] I was using flutterflow from multiple platforms (e.g. both web and MacOS Desktop app)

### Relative to the time the changes were made, data was lost within

- [ ] 1 minute
- [ ] 1 hour
- [ ] 1 day
- [ ] 1 week

### When following my steps to reproduce, data loss happens

- [ ] ~100% of the time
- [ ] ~50% of the time
- [ ] ~10% of the time
- [ ] ~1% of the time
- [ ] Very rarely or only encountered once

**Comment by myElectrical** [comment]
As an update, I have disabled the navigate back button on all the navigation links, which appears to stop the multiple API calls. I'm not sure how this will affect things going forward.

**Comment by msusviela** [comment]
Hi @myElectrical. Thanks for submitting the issue. I've taken a look at your project, but with a lot going on, it's challenging to pinpoint the issue. Given the nature of the problem you're highlighting, we need a blank project with the issue steps set up in it. Without these details, it might be a bit challenging for us to investigate the issue. Please provide the required information. If the issue is already resolved for you, please feel free to close the issue. Thanks!

**Comment by myElectrical** [comment]
Hi, thanks for getting back to me.  I had tried to place the API calls as close as possible to where they were used but high enough in the widget train to be only called as necessary.  Since then I have moved them to the on-page load event, and turned off back navigation which appears to have reduced the number of calls.

For the time being I will stick with this solution and close the issue. If my work around turns out not to have been that great I can reopen another issue at that tine.
